### PR TITLE
Do not rely on frontend to pass filereference

### PR DESCRIPTION
**Pull request by jbout**
Do not pass the reference to the compiled file on the filesystem to the client and then back to the server.

depends on:
* https://github.com/oat-sa/extension-tao-delivery/pull/476
* https://github.com/oat-sa/extension-tao-delivery-rdf/pull/327

### CODE REVIEW

File: models/classes/runner/QtiRunnerService.php [hunk 1]
```diff
@@ -76,6 +76,7 @@
 use qtism\data\AssessmentItemRef;
 use qtism\runtime\tests\SessionManager;
 use oat\libCat\result\ResultVariable;
+use League\Flysystem\FileNotFoundException;
```
Comment (siwane) [inline models/classes/runner/QtiRunnerService.php:79]: This is not the exception caught before

Comment (jbout) [reply models/classes/runner/QtiRunnerService.php:79]: it's actually the exception caught afterwards, but I believe neither can be triggered

**Comment by siwane** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by siwane** [comment]
unit tests are breaking

**Comment by jbout** [comment]
Unit tests fixd by merging develop

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**